server/router/src/main/java/org/infinispan/server/router/RoutingTable.java

```diff
@@ -0,0 +1,44 @@
+package org.infinispan.server.router;
+
+import org.infinispan.server.router.routes.Route;
+import org.infinispan.server.router.routes.RouteDestination;
+import org.infinispan.server.router.routes.RouteSource;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Stream;
+
+public class RoutingTable {
+
+    private Set<Route<? extends RouteSource, ? extends RouteDestination>> routes = new HashSet<>();
+
+    public RoutingTable(Set<Route<? extends RouteSource, ? extends RouteDestination>> routes) {
+        this.routes.addAll(routes);
+    }
+
+    public int routesCount() {
+        return routes.size();
+    }
+
+    public Stream<Route<? extends RouteSource, ? extends RouteDestination>> streamRoutes() {
+        return routes.stream();
+    }
+
+    @SuppressWarnings("unchecked")
+    public <Source extends RouteSource, Destination extends RouteDestination>
```

Comment:
Do you need the types here?  Can't you just use `? extends RouteSource` in the method params for example?